### PR TITLE
Fix PEM armor for EC private keys when encoding

### LINKED ISSUE #875: jwk.EncodePEM, jwk.DecodePEM not reciprocal for ECDSA private keys
**Describe the bug**

It is currently not possible to take the output of `jwk.EncodePEM()` and give it to `jwk.ParseKey(key, jwk.WithPEM(true))` when the input to `jwk.EncodePEM` is a `*ecdsa.PrivateKey`.

This is because `EncodeX509` uses `ECDSA PRIVATE KEY` as its PEM armor
https://github.com/lestrrat-go/jwx/blob/defd35c63b7f77ceb29d1788de87c093b1a63a73/jwk/jwk.go#L272
and `DecodePEM` expects `EC PRIVATE KEY`
https://github.com/lestrrat-go/jwx/blob/defd35c63b7f77ceb29d1788de87c093b1a63a73/jwk/jwk.go#L331

**To Reproduce / Expected behavior**

``` golang
func TestECDSAPEM(t *testing.T) {
	// go make an EC key at https://mkjwk.org/
	key, err := jwk.ParseKey([]byte(`{
		"kty": "EC",
		"d": "zqYPTs5gMEwtidOqjlFJSk6L4BQSfhCJX6FTgbuuiE0",
		"crv": "P-256",
		"x": "AYwhwiE1hXWdfwu-HlBSsY5Chxycu-LyE6WsZ_w2DO4",
		"y": "zumemGclMFkimMsKMXlLdKYWtLle58e4N9hDPcN7lig"
	}`))
	if err != nil {
		t.Fatal(err)
	}
	pem, err := jwk.EncodePEM(key)
	if err != nil {
		t.Fatal(err)
	}
	_, err = jwk.ParseKey(pem, jwk.WithPEM(true))
	if err != nil {
		t.Fatal(err)
	}
}
```

### PR DESCRIPTION
**Pull request by lestrrat**
fixes #875

### CODE REVIEW

**Comment by lestrrat** [comment]
TODO: before merging, I'd like to do a bit of code cleanup for other related code (use more constants, etc)

File: jwk/jwk_test.go [hunk 1]
```diff
@@ -2070,3 +2070,25 @@ func TestGH730(t *testing.T) {
...
+		"y": "zumemGclMFkimMsKMXlLdKYWtLle58e4N9hDPcN7lig"
+	}`))
+	if err != nil {
+		t.Fatal(err)
+	}
+	pem, err := jwk.EncodePEM(key)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = jwk.ParseKey(pem, jwk.WithPEM(true))
```
Comment (mheggeseth) [inline jwk/jwk_test.go:2090]: **suggestion:** In addition to verifying there's no error, verify this key is equivalent to the key above.

File: jwk/jwk_test.go [hunk 2]
```diff
@@ -2070,3 +2070,25 @@ func TestGH730(t *testing.T) {
 	require.NoError(t, set.AddKey(key), `first AddKey should succeed`)
 	require.Error(t, set.AddKey(key), `second AddKey should fail`)
 }
+
+func TestECDSAPEM(t *testing.T) {
```
Comment (mheggeseth) [inline jwk/jwk_test.go:2074]: **suggestion (non-blocking):** Include a similar reciprocal PEM encode/decode test for each type of JWK you support.

**Review by mheggeseth — CHANGES_REQUESTED** [review]
LGTM. A couple minor changes requested.

File: jwk/jwk_test.go (hunk 1 above)
Comment (lestrrat) [reply jwk/jwk_test.go:2090]: that's exactly what I intend to do

**Comment by lestrrat** [comment]
@mheggeseth I reused an existing test to check for PEM encodings as well, going to merge

**Pull request merged**